src/rbd.cc

```diff
@@ -1368,6 +1368,56 @@ static void update_snap_name(char *imgname, char **snap)
     *snap = s;
 }
 
+static bool is_valid_name(const char *name)
+{
+   char ch;
+   unsigned int i; 
+
+   if(name == NULL)
+     return true;
+
+   if(strlen(name) == 0)
+     return false;
+   
+   for (i = 0; i < strlen(name); i++ ) {
+      ch = name[i];
+      if ( !( ( ch >= '0' && ch <= '9' ) ||
```

Comment:
With due respect to you.
I feel there should be a restriction on the name for pool/Image name/snap name, I feel in pool creation (ie ceph) also we need to have a restriction.
Many characters do give problems, Say if user gives 
ubuntu@Host-CephAdmin$ ceph osd pool create $#$#* 128 128
pool '00*' created

This will create pool 00* here $#$# gets interpreted differently.

Some of the special characters the shell interprets it differently say &, user should be not allowed using such characters in names. 

I believe Pool name need not be utf-8, since all other ceph commands are Ascii in nature hence pool name which is used by the product for its functioning can be Ascii only. Currently by allowing various other special characters it can create problems one such we saw above.
fs stores data so it has to be utf-8 compatible as user can have its data in any language.

Please correct me if my understanding is wrong.